File: TypeScript/6.Z 字形变换.ts
function convert(s: string, numRows: number): string {
    const t = new Array<string>(numRows).fill("")
    let ans: string = ""
    let i: number = 0
    while (i < s.length) {
        for (let j = 0; j < numRows && i < s.length; ++j) {
            t[j] += s[i++]
        }
        for (let k = numRows - 2; k > 0 && i < s.length; --k) {
            t[k] += s[i++]
        }
    }
    for (const x of t) {
        ans += x
    }
    return ans
};

File: Python/6.Z 字形变换.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        t: list[str] = [""] * numRows
        ans: str = ""
        i: int = 0
        while i < len(s):
            for j in range(0, numRows, 1):
                if not i < len(s):
                    break
                t[j] = t[j] + s[i]
                i += 1
            for k in range(numRows-2, 0, -1):
                if not i < len(s):
                    break
                t[k] = t[k] + s[i]
                i += 1
        for x in t:
            ans += x
        return ans
